parquet-column/src/main/java/org/apache/parquet/column/values/bloomfilter/BlockSplitBloomFilter.java

```diff
@@ -379,4 +379,32 @@ public long hash(float value) {
   public long hash(Binary value) {
     return hashFunction.hashBytes(value.getBytes());
   }
+
+  private boolean checkCompatibility(BlockSplitBloomFilter that) {
+    return this.maximumBytes == that.maximumBytes &&
+      this.hashStrategy == that.hashStrategy &&
+      this.getAlgorithm() == that.getAlgorithm();
+  }
+
+  @Override
+  public BloomFilter union(BloomFilter bloomFilter) {
+    if (bloomFilter instanceof BlockSplitBloomFilter) {
+      BlockSplitBloomFilter that = (BlockSplitBloomFilter) bloomFilter;
+
+      if (!checkCompatibility(that)) {
+        throw new RuntimeException("Bloom filters are not compatible! Please check"
+          + "if they have the same maximum Bytes, hash strategy and algorithm.");
```

Comment:
I would suggest using `IllegalArgumentException` instead.

I think, the message would be more useful if we add the actual values of max bytes, hash strategy etc. instead of suggesting to check them. When debugging from log files this message would not help. Maybe, `checkCompatibility` itself can throw the specific exception so you would know which value is incorrect and what would be the correct one.

Reply:
You re right! Let me fix it.